docs(@clayui/core): adds documentation of selection types and "manual" selection (#4626)
liferay/clay: packages/clay-core/docs/treeview.mdx

</TreeView>
```

## Selection

The selection allows the user to select one or more items in the TreeView, there are three main interactions that can be configured and are defined using the `selectionMode` prop.

-   `single` select only one item.
-   `multiple` select multiple items.

> -   `multiple-recursive` selects multiple items and the item's children recursively. The Parent item is also selected if all children are selected, if not all selected it is marked as intermediate.

Reviewer: ```suggestion
`multiple-recursive` selects multiple items and the item's children recursively. When all children are selected, the parent will be marked as selected, otherwise it will be marked as intermediate.
```

> The selection can be configured and composed in different ways depending on each use case, setting the `selectionMode` and declaring the `<Checkbox />` in the item when selecting the checkbox will respect the configuration of the selection. It is also possible to make the "manual" selection to customize who will trigger the selection or modify the look of the selected state, see more information in the [manual selection](#manual-selection) section.

Reviewer: ```suggestion
The selection can be configured and composed in different ways depending on each use case. Setting the `selectionMode` and declaring the `<Checkbox />` in the item when selecting the checkbox will respect the configuration of the selection. It is also possible to configure the selection manually to customize what will trigger the selection or modify the look of the selected state. For more information check the [manual selection](#manual-selection) section.
```

@matuzalemsteles the only part I don't fully understand is this

> Setting the `selectionMode` and declaring the `<Checkbox />` in the item when selecting the checkbox will respect the configuration of the selection.

I know that these things are hard to explain, so let's have a quick talk/call about it on Slack, if you think it's worth it.

		<TreeView
			onSelectionChange={setSelectionChange}
			selectedKeys={selectedKeys}
			selectionMode="multiple"
		>
			{...}
		</TreeView>
	);
}
```

### Recursive Multiple Selection

> By default, when selecting an item with nested items, the item's children are recursively selected.

Reviewer: ```suggestion
By default, when selecting an item with nested items, its children are recursively selected.
```
